lib/_http_client.js

```diff
@@ -30,10 +30,17 @@ function ClientRequest(options, cb) {
 
   var agent = options.agent;
   var defaultAgent = options._defaultAgent || Agent.globalAgent;
+
+  if (agent != null && agent !== false && !(agent instanceof Agent.Agent)) {
+    throw new Error(
+       'Agent option must be an instance of http.Agent, undefined or false. ' +
+       `Received type: ${typeof agent} instead`
+    );
+  }
+
   if (agent === false) {
     agent = new defaultAgent.constructor();
-  } else if ((agent === null || agent === undefined) &&
-             typeof options.createConnection !== 'function') {
+  } else if (agent == null && typeof options.createConnection !== 'function') {
     agent = defaultAgent;
   }
```

Comment:
Instead of duplicating all the checks on line 34, couldn't you just add:

```js
} else if (!(agent instanceof Agent.Agent)) {
  throw new TypeError(...);
}
```